src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -56,21 +57,30 @@ static zns_sm_metadata_t make_metadata(
   
   auto config_segment_size = get_conf<Option::size_t>(
     "seastore_segment_size");
-  INFO("config segment size: {}", config_segment_size);
-  size_t zones_per_segment = config_segment_size / zone_capacity;
+
+  size_t segment_size = (zone_size == 0) ? config_segment_size : (zone_size << SECT_SHIFT);
+  size_t zones_per_segment = segment_size / (zone_size << SECT_SHIFT);
   size_t segments = (num_zones - 1) * zones_per_segment;
   
-  INFO("size {}, block_size {}, allocated_size {}, configured_size {}, "
-    "segment_size {}",
-    data.size,
+  INFO(
+    "\n\t device size {}, block_size {}, allocated_size {}, config_size {},"
+    "\n\t config_segment_size {}, total zones {}, zone_size {}, zone_capacity {},"
+    "\n\t total segments {}, zones per segment {}, segment size {}",
+    size,
     data.block_size,
     data.allocated_size,
     config_size,
-    config_segment_size);
+    config_segment_size,
+    num_zones,
+    zone_size << SECT_SHIFT,
+    zone_capacity << SECT_SHIFT,
+    segments,
+    zones_per_segment,
+    (zone_capacity << SECT_SHIFT) * zones_per_segment);
   
   zns_sm_metadata_t ret = zns_sm_metadata_t{
     size,
-    config_segment_size,
+    segment_size,
     zone_capacity * zones_per_segment,
```

Comment:
Is it true that for an empty zone, zone_capacity can be smaller than the zone_size?

Reply:
Zone Size / Zone capacity is what the SSD reports and is not configurable by user.
ZNS spec supports that zone capacity can be less than or equal to zone size.
So if the SSD manufacturer has zone size and capacity set to different values, then yes it is possible for zone capacity to be less than zone size. But zone state (empty/open/closed/full) does not have anything to do with it, i.e. zone capacity will be same in all valid zone states. 
For ex: 64MB zone size and 48MB zone capacity is a valid scenario on ZNS device in all zone states. Writes and reads cannot be done in every zone between 48MB and 64MB, it is essentially a hole.

Once I have the Seastore working on a zns device with zone-size == zone-capacity, I will look into adding support for different size and capacity.

Reply:
OK, so zone_size doesn't seem to be useful from SeaStore perspective.